// ruru@2.0.0-beta.13/dist/server.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ruruHTML = exports.defaultHTMLParts = exports.makeHTMLParts = void 0;
const tslib_1 = require("tslib");
const fs_1 = require("fs");
const path = tslib_1.__importStar(require("path"));
const { version } = JSON.parse((0, fs_1.readFileSync)(path.resolve(__dirname, `../package.json`), "utf8"));
// ENHANCE: make this 'readFileSync' call webpackable
const graphiQLContent = (0, fs_1.readFileSync)(path.resolve(__dirname, "../bundle/ruru.min.js"), "utf8");
// Ref: https://v8.dev/features/subsume-json
const escapeJS = (str) => {
    return str
        .replaceAll("<!--", "<\\!--")
        .replaceAll("</script", "<\\/script")
        .replaceAll("<script", "<\\script");
};
const baseMetaTags = `\
<meta charset="utf-8" />`;
const baseTitleTag = `\
<title>Ruru - GraphQL/Grafast IDE</title>
`;
const baseStyleTags = `\
<style>
body {
  margin: 0;
}
#ruru-root {
 height: 100vh;
}
</style>`;
const baseHeaderScripts = `\
<script src="https://unpkg.com/prettier@1.13.0/standalone.js"></script>
<script src="https://unpkg.com/prettier@1.13.0/parser-graphql.js"></script>`;
const baseElements = `\
<div id="ruru-root"></div>`;
const baseBodyScripts = `\
<script src="https://cdn.jsdelivr.net/npm/mermaid@9.4.3"></script>
<script>/*! For license information, see https://unpkg.com/ruru@${version}/bundle/ruru.min.js.LICENSE.txt */
${escapeJS(graphiQLContent)}</script>`;
const baseBodyInitScript = `\
<script>
  const { React, createRoot, Ruru } = RuruBundle;
  const tree = React.createElement(Ruru, RURU_CONFIG);
  const container = document.getElementById("ruru-root");
  const root = createRoot(container);
  root.render(tree);
</script>`;
function makeHTMLParts() {
    return {
        metaTags: baseMetaTags,
        titleTag: baseTitleTag,
        styleTags: baseStyleTags,
        headerScripts: baseHeaderScripts,
        bodyContent: baseElements,
        bodyScripts: baseBodyScripts,
        bodyInitScript: baseBodyInitScript,
    };
}
exports.makeHTMLParts = makeHTMLParts;
exports.defaultHTMLParts = Object.freeze(makeHTMLParts());
function ruruHTML(config, htmlParts = exports.defaultHTMLParts) {
    const { metaTags, titleTag, styleTags, headerScripts, bodyContent, bodyScripts, bodyInitScript, } = htmlParts;
    return `\
<!DOCTYPE html>
<html lang="en">
<head>
${metaTags}
${titleTag}
${styleTags}
<script>const RURU_CONFIG = ${escapeJS(JSON.stringify(config))};</script>
${headerScripts}
</head>
<body>
${bodyContent}
${bodyScripts}
${bodyInitScript}
</body>
</html>`;
}
exports.ruruHTML = ruruHTML;
